testing/state-transition/state-transition.go

```diff
@@ -58,23 +58,30 @@ type (
 	TestStateProcessorT          = core.StateProcessor
 )
 
-type testKVStoreService struct {
-	ctx sdk.Context
```

Comment:
this was a first attempt at creating state but it is wrong.
We should use the context provided by the transactions and extract the state frorm there